src/language-js/print/module.js

```diff
@@ -246,19 +246,36 @@ function shouldNotPrintSpecifiers(node, options) {
   );
 }
 
-function printImportAssertions(path, options, print) {
+/**
+ * Print Import Attributes syntax.
+ * If old ImportAssertions syntax is used, print them here.
+ */
+function printImportAttributes(path, options, print) {
   const { node } = path;
-  if (!isNonEmptyArray(node.assertions)) {
-    return "";
+
+  // For @babel/parser
+  const keyword = node.extra?.deprecatedAssertSyntax ? "assert" : "with";
+  if (isNonEmptyArray(node.attributes)) {
```

Comment:
Nit, but you can move the keyword check inside the if statement
```suggestion
  if (isNonEmptyArray(node.attributes)) {
    const keyword = node.extra?.deprecatedAssertSyntax ? "assert" : "with";
```